### PR TITLE
Add "open webpage" row to student work detail

### LINKED ISSUE #3169: Add “open webpage” row to student work detail
Rather than make someone hit the share button, it might be helpful to provide the original webpage as part of the detail listing

### PR DESCRIPTION
**Pull request by drewvolz**
Closes #3169

* Introduces a function to create a full job URL from the id
* Shows a pushbutton on iOS to open a webpage
* Shows a card with text on Android to open a webpage

iOS | Android
--|--
screen shot 2018-11-17 at 2 23 28 pm | android-jobs

### CODE REVIEW

**Comment by hawkrives** [comment]
I don't know that I like the wording "Open original posting webpage". 

I don't have anything better though. 

I also don't think that the Android one feels like a thing you can interact with, but I'll defer to Kris on that I think.

**Comment by drewvolz** [comment]
> I don't know that I like the wording "Open original posting webpage". I don't have anything better though.

I feel similarly, and can't think of an alternative.

> I also don't think that the Android one feels like a thing you can interact with, but I'll defer to Kris on that I think.

That's our current convention with android links... is text inside a card. Would you prefer the url to be visible instead?

**Review by rye — COMMENTED** [review]
I may do a more thorough code review, but towards Hawken's suggestion I think the wording should be "Open posting".

On Android, I would honestly prefer to do away with the card and use a FAB. I also think the usage of the `open` ionicon would help a lot. Maybe one or the other, or both.

**Comment by rye** [comment]
I think a FAB is an excellent style element that lots of Android users anticipate, but we don't use.

**Comment by hawkrives** [comment]
Okay, but isn't a FAB intended for the "primary" action in a view?

Although I'm not sure what other actions you would take from the view…

**Comment by rye** [comment]
@hawkrives, you're exactly right.

From [the Material spec](https://material.io/design/components/buttons-floating-action-button.html):

>A FAB promotes an important, constructive action such as:
>- Create
>- Favorite
>- Share
>- Start a process

I think of opening the webpage as "starting a process".  Of course, for strange job listings like the lovely one that we have here, people should instead be going to the google form, but I digress. Ideally, the Student Work postings page would adequately render on a phone so that people can get to the application through that page.

Since this view is basically just a representation of the data on a web page, I think having a FAB with the `open` icon would not be considered a minor action in this view.

Like this, but round…:

FAB with icon-open, but rectangular

I think anything short of a button for this is kinda strange on Android.

Also, can someone remind me why links don't look like links?

**Comment by drewvolz** [comment]
> Also, can someone remind me why links don't look like links?

We have that ability on iOS thanks to TextInput but we don’t have a natural link parser for android

File: source/views/sis/student-work/detail.android.js [hunk 1]
```diff
@@ -165,6 +165,20 @@ function Timeline({job}: {job: JobType}) {
...
 
+function OpenWebpage({job}: {job: JobType}) {
+	return job.id ? (
+		<Card header="Webpage" style={styles.card}>
+			<Text
+				key={job.id}
+				onPress={() => openUrl(createJobFullUrl(job))}
+				style={styles.cardBody}
+			>
+				Open original posting webpage
```
Comment (hawkrives) [inline source/views/sis/student-work/detail.android.js]:
```suggestion
				Open Posting
```

File: source/views/sis/student-work/detail.ios.js [hunk 2]
```diff
@@ -151,6 +158,17 @@ function Timeline({job}: {job: JobType}) {
 	) : null
 }
 
+function OpenWebpage({job}: {job: JobType}) {
+	return job.id ? (
+		<Section header="">
+			<PushButtonCell
+				onPress={() => openUrl(createJobFullUrl(job))}
+				title="Open original posting webpage"
```
Comment (hawkrives) [inline source/views/sis/student-work/detail.ios.js]:
```suggestion
				title="Open Posting"
```

**Review by hawkrives — CHANGES_REQUESTED** [review]
I think we're OK with not doing a FAB in this PR; that will be a much more significant change.

Let's file an issue and do the FAB in another PR, as much as I want it.

File: source/views/sis/student-work/detail.ios.js (hunk 2 above)
Comment (rye) [reply source/views/sis/student-work/detail.ios.js]: Downcase second word?

Comment (hawkrives) [reply source/views/sis/student-work/detail.ios.js]: On iOS, at least, these are usually in title case

**Comment by rye** [comment]
@hawkrives I am okay with doing FABs elsewhere. I will admit that they are out of scope since we never have done them and I think that'll require an added dependency.

**Review by hawkrives — APPROVED** [review]

**Review by rye — APPROVED** [review]

**Pull request merged**